Lesson18/gozde/script.js

```diff
@@ -0,0 +1,223 @@
+
+
+console.log("Ex4" );
+console.log('----------------------------');
+// 5. Write a function that takes an array of integers and returns a new array with even numbers only.
+// Input: [1, 2, 3, 4, 7, 10]
+// Output: [1, 2, 4, 10]
+const testArrayEx5 = [1, 2, 3, 4, 7, 10];
+
+function filterEvenNumbers(numbersArray) {
+  let evenNumbers = [];
+  
+  for (let i = 0; i < numbersArray.length; i++) {
+      if (numbersArray[i] % 2 === 0) {
+          evenNumbers.push(numbersArray[i]);
+      }
+  }
+  return evenNumbers;
+}
+
+const testArrayEx5 = [1, 2, 3, 4, 7, 10];
+const evenNumbersArray = filterEvenNumbers(testArrayEx5);
+console.log(evenNumbersArray); // Output: [2, 4, 10]
+
+
+
+console.log('Exercise 5: ');
+console.log('----------------------------');
+// 6. Write a function that takes an array and a value, and returns a new array with all occurrences of the value removed.
+// Input: ([1, 2, 3, 4, 2], 2)
+// Output: [1, 3, 4]
+const testArrayEx6 = [1, 2, 3, 4, 2];
+
+/*
+1. write a function with array and values as 2 parameters
+2. create placeholder variable to store new array
+3. loop through every values of array
+4. use if / else condition to find if current value is equal to the value param
+5. return the result
+*/
+function removeValueFromArray(array, valueToRemove){
+  let newArray = [];
+
+  for(let i = 0; i < array.length; i++) {
+    console.log(i, array[i], valueToRemove);
+    if (array[i] !== valueToRemove){
+      console.log('inside if statemnt', i, array[i], valueToRemove);
+      newArray.push(array[i]); 
+      // 'a' !== 'b' => ['a']
+      // 'b' === 'b' => ['a']
+      // 'c' !== 'b' => ['a', 'c']
+    }
+  }
+
+  return newArray;
+}
+
+console.log("exercise 6:",removeValueFromArray(testArrayEx6, 3), removeValueFromArray(['a', 'b', 'c'], 'b'));
+console.log('----------------------------');
+// 7. Write a function that takes an object and returns the keys as an array.
+// Input: {name: "John", age: 30, city: "New York"}
+// Output: ["name", "age", "city"]
+const testObjEx7 = {name: "John", age: 30, city: "New York", isStudent: false};
+
+console.log(Object.keys(testObjEx7));
+
+console.log("exercise 7:");
+console.log('----------------------------');
+// 8. Write a function that takes an object and returns the values as an array.
+// Input: {name: "John", age: 30, city: "New York"}
+// Output: ["John", 30, "New York"]
+const testObjEx8 = {name: "John", age: 30, city: "New York"};
+
+console.log("exercise 8:", Object.values(testObjEx8));
+console.log('----------------------------');
+// 9. Write a function that takes an object and a key, and returns true if the key exists in the object, otherwise false.
+// Input: ({name: "John", age: 30, city: "New York"}, "age")
+// Output: true
+// Input: ({name: "John", age: 30, city: "New York"}, "surname")
+// Output: false
+const testObjEx9 = {name: "John", age: 30, city: "New York", number: undefined};
+
+function ex9(object, key) {
```

Comment:
What you used here is a for...in loop that allows looping through objects in JS: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in
However, you need to use a different built-in function here:
`object.hasOwnProperty(key)` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty